### DISCUSSION TITLE
Ariadne codegen with custom HTTP client

### DISCUSSION THREAD

**Discussion by dkbarn** [mirumee/ariadne#1069] [Q&A]
The README for ariadne-codegen mentions that you can pass a custom HTTP client for complex scenarios, but it doesn't provide any guidance beyond the one-line code snippet:

```python
client = Client(http_client=CustomComplexHttpClient())
```

Where can I find documentation or examples of how to implement such a CustomComplexHttpClient class? Presumably it needs to conform to some sort of interface. Where is that interface defined? What methods need to exist on this class?

**Comment by rafalp** [comment]
Ariadne Codegen uses [HTTPX](https://github.com/encode/httpx). We should update readme to mention that.
